src/main/java/com/capitalone/dashboard/event/sync/SyncDashboard.java

```diff
@@ -79,10 +85,24 @@ public Widget getWidget(String name, Dashboard dashboard) {
      */
     private void addCollectorItemToDashboard(List<Dashboard> existingDashboards, CollectorItem collectorItem, CollectorType collectorType, boolean addWidget) {
         if (CollectionUtils.isEmpty(existingDashboards)) return;
-        /**
-         * The assumption is the dashboard already has a SCM widget and the sync process should not add SCM widgets.
-         */
-        if(CollectorType.SCM.equals(collectorType)) return;
+        Iterable<FeatureFlag> ff = featureFlagRepository.findAll();
+        if(!IterableUtils.isEmpty(ff)){
+            FeatureFlag f = ff.iterator().next();
+            if(CollectorType.SCM.equals(collectorType) && f.isScm()) return;
+            if(CollectorType.CodeQuality.equals(collectorType) && f.isCodeQuality()) return;
+            if(CollectorType.LibraryPolicy.equals(collectorType) && f.isLibraryPolicy()) return;
+            if(CollectorType.StaticSecurityScan.equals(collectorType) && f.isStaticSecurity()) return;
+            if(CollectorType.Artifact.equals(collectorType) && f.isArtifact()) return;
+            if(CollectorType.Build.equals(collectorType) && f.isBuild()) return;
+            if(CollectorType.Deployment.equals(collectorType) && f.isDeployment()) return;
+            if(CollectorType.AgileTool.equals(collectorType) && f.isAgileTool()) return;
+            if(CollectorType.Test.equals(collectorType) && f.isTest()) return;
+        }else{
+            /**
+             * The assumption is the dashboard already has a SCM widget and the sync process should not add SCM widgets.
+             */
+            if(CollectorType.SCM.equals(collectorType)) return;
+        }
```

Comment:
SCM will be discovered if f.isScm is false and some other flag is already true?

Reply:
@benj58xu  addCollectorItemToDashboard() is called for each collector type, there will be one condition checked at a time.